Dev

flo7842 opened the pull request:
This branch contain all files of the application. Interfaces, Implementations, Factories and the configuration of Dependency Injection.

flo7842 committed:
Creation of models and their implementations

flo7842 committed:
Creating Provider, adding exceptions types to interfaces

flo7842 committed:
test

flo7842 committed:
test

flo7842 committed:
Create interface

flo7842 committed:
Create factories

flo7842 committed:
Create services

flo7842 committed:
Create interfaces

flo7842 committed:
Create Injection

flo7842 committed:
Create Startup Class for Dependency Injection

flo7842 committed:
Update logger config file

flo7842 committed:
Update factories files

flo7842 committed:
Create Implementations of interface

flo7842 committed:
Create and update of interfaces

flo7842 committed:
Merge branch 'wip/Interface' into dev

flo7842 committed:
Create Logger

flo7842 committed:
Create README

flo7842 committed:
Create README

flo7842 committed:
Merge remote-tracking branch 'origin/dev' into dev

flo7842 committed:
Remove test file

flo7842 committed:
Remove test file

flo7842 committed:
Merge remote-tracking branch 'origin/dev' into dev

flo7842 committed:
Removing unused methods


Changing methods name

flo7842 committed:
Applying the C# naming convention for namespaces

greg7859 commented on edc-client-dotnet/NLog.config:
```diff
@@ -0,0 +1,22 @@
+		<target name="file" 
+		        xsi:type="File" 
+		        createDirs="true"
+		        fileName="${basedir}\logs\debug.txt" 
```
Generally, the file is named XXX.log where XXX is the application name. In our case, the file should be named: client.log

greg7859 commented on edc-client-dotnet/internalImpl/DocumentationManagerImpl.cs:
```diff
@@ -0,0 +1,66 @@
+        private IEdcReader _reader;
+        private IKeyUtil _keyUtil;
+        private Dictionary<String, IContextItem>? _contexts;
+        private static NLog.Logger _logger = NLog.LogManager.GetCurrentClassLogger();
```
Add  NLog in the **using** section

```csharp
using NLog;
`
// ...

private static Logger _logger = LogManager.GetCurrentClassLogger();
```

greg7859 commented on edc-client-dotnet/internalImpl/EdcClientImpl.cs:
```diff
@@ -0,0 +1,131 @@
+        private readonly ITranslationManager _translationManager;
+        private readonly IInformationManager _informationManager;
+        private readonly IUrlUtil _urlUtil;
+        private static readonly NLog.Logger _logger = NLog.LogManager.GetCurrentClassLogger();
```
Same remark as DocumentationManagerImpl

greg7859 commented on edc-client-dotnet/internalImpl/Constants.cs:
```diff
@@ -0,0 +1,7 @@
+{
+    public class Constants
+    {
+        public const String SERVER_URL = "https://demo.easydoccontents.com";
```
Only for the test so move it on the test project

greg7859 commented on edc-client-dotnet/factory/ContextItemFactory.cs:
```diff
@@ -0,0 +1,20 @@
+
+        public IContextItem Create()
+        {
+            return (IContextItem)_serviceProvider.GetService(typeof(ContextItemImpl));
```
Either you use the injector
```csharp
return (IContextItem)_serviceProvider.GetService(typeof(IContextItem));
```

or you create an instance with the implementation class

```csharp
return new ContextItemImpl();
```

It's not a good idea to mix both

greg7859 commented on edc-client-dotnet/model/IContextItem.cs:
```diff
@@ -0,0 +1,44 @@
+namespace edc_client_dotnet.model
+{
+    // This class is a specialized DocumentationItem for the contextual content (bricks)
+    public interface IContextItem : IDocumentationItem
```
In .NET, you have the Properties. Use it and avoid to reproduce the java source code.

See: https://docs.microsoft.com/fr-fr/dotnet/csharp/programming-guide/classes-and-structs/properties and https://docs.microsoft.com/fr-fr/dotnet/csharp/programming-guide/classes-and-structs/interface-properties

greg7859 commented on edc-client-dotnet-test/EdcClientSingletonTest.cs:
```diff
@@ -0,0 +1,102 @@
+    [TestInitialize]
+    public void Setup()
+    {
+        EdcClientSingleton.GetInstance().SetServerUrl("https://demo.easydoccontents.com");
```
Why did you not use the Constant?
Either you define a constant and you use it or you write independant unit tests.

flo7842 committed:
Update factories and configuration of injection container

flo7842 committed:
Using naming convention for accessors and mutators

greg7859 reviewed (changes requested):
Need to rename all namespaces

Namespaces are named using [Pascal Case](https://en.wikipedia.org/wiki/Pascal_Case) (also called UpperCamelCase) with no underscores. This means the first letter of every word in the name is capitalized. For example: MyNewNamespace. Pascal Case also means that acronyms of three or more letters should only have the first letter capitalized (MyXmlNamespace instead of MyXMLNamespace).

flo7842 committed:
Removing unused methods


Changing methods name

Applying the C# naming convention for namespaces

Update factories and configuration of injection container

Using naming convention for accessors and mutators

Edit tests

flo7842 committed:
Merge remote-tracking branch 'origin/dev' into dev
